nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -0,0 +1,234 @@
+                                      target_shape=shape[:3],
+                                      interpolation='nearest')
+        mask, _ = masking._load_mask_img(mask_img)
+    signals = np.empty((shape[3], n_seeds))
+    # Create an array of shape (3, array.shape) containing the i, j, k indices
+    # in voxel space
+    coords = np.vstack((np.indices(shape[:3]),
+                        np.ones((1,) + shape[:3])))
+    # Transform the indices into native space
+    coords = np.tensordot(affine, coords, axes=[[1], [0]])[:3]
+    for i, seed in enumerate(seeds):
+        seed = np.asarray(seed)
+        # Compute square distance to the seed
+        dist = ((coords - seed[:, None, None, None]) ** 2).sum(axis=0)
+        if radius is None or radius ** 2 < np.min(dist):
+            dist_mask = (dist == np.min(dist))
+        else:
+            dist_mask = (dist <= radius ** 2)
+        if mask_img is not None:
+            dist_mask = np.logical_and(mask, dist_mask)
+        if not dist_mask.any():
+            raise ValueError('Seed #%i is out of the mask' % i)
+        signals[:, i] = np.mean(niimg.get_data()[dist_mask], axis=0)
+    return signals
+
+
+def _compose_err_msg(msg, **kwargs):
+    """Append key-value pairs to msg, for display.
+
+    Parameters
+    ==========
+    msg: string
+        arbitrary message
+    kwargs: dict
+        arbitrary dictionary
+
+    Returns
+    =======
+    updated_msg: string
+        msg, with "key: value" appended. Only string values are appended.
+    """
+    updated_msg = msg
+    for k, v in kwargs.iteritems():
+        if isinstance(v, basestring):
+            updated_msg += "\n" + k + ": " + v
+
+    return updated_msg
+
+
+class NiftiSpheresMasker(BaseEstimator, TransformerMixin, CacheMixin):
+    """Class for masking of Niimg-like objects using seeds.
+
+    NiftiSpheresMasker is useful when data from given seeds should be
+    extracted. Use case: Summarize brain signals from seeds that were
+    obtained from prior knowledge.
+
+    Parameters
+    ==========
+    seeds: List of triplet of coordinates in native space
+        Seed definitions. List of coordinates of the seeds in the same space
+        as the images (typically MNI or TAL).
+
+    radius: float, optional.
+        Indicates, in millimeters, the radius for the sphere around the seed.
+        Default is None (signal is extracted on a single voxel).
+
+    smoothing_fwhm: float, optional
+        If smoothing_fwhm is not None, it gives the full-width half maximum in
+        millimeters of the spatial smoothing to apply to the signal.
+
+    standardize: boolean, optional
+        If standardize is True, the time-series are centered and normed:
+        their mean is set to 0 and their variance to 1 in the time dimension.
+
+    detrend: boolean, optional
+        This parameter is passed to signal.clean. Please see the related
+        documentation for details
```

Comment:
This has been copied from NiftiLabelsMasker, we should open another issue about that and fix it afterward.